### PR TITLE
[Az.Resources] Formatted the output format of DeploymentVariable

### PR DESCRIPTION
**Pull request by BethanyZhou**
## Description
Fixing: https://github.com/Azure/azure-powershell/issues/16523
Before:

```
Id                      : /subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/providers/Microsoft.Resources/deployments
                          /test
DeploymentName          : test
Location                : westeurope
ProvisioningState       : Succeeded
Timestamp               : 11/30/2021 2:06:54 AM
Mode                    : Incremental
TemplateLink            :
Parameters              :
                          Name             Type                       Value
                          ===============  =========================  ==========
                          stuff            Array                      [
                            "one",
                            "two"
                          ]

Outputs                 :
                          Name             Type                       Value
                          ===============  =========================  ==========
                          stuff            Array                      [
                            "one",
                            "two"
                          ]

DeploymentDebugLogLevel :
```

After:
```
PS C:\AzurePowershell\azure-powershell\artifacts\Debug\Az.Resources> $result

DeploymentName          : testarray
ResourceGroupName       : bez-rg
ProvisioningState       : Succeeded
Timestamp               : 12/9/2021 3:26:31 AM
Mode                    : Incremental
TemplateLink            :
Parameters              :
                          Name             Type                       Value
                          ===============  =========================  ==========
                          stuff            Array                      [
                                                                        "one",
                                                                        "two"
                                                                      ]


Outputs                 :
                          Name             Type                       Value
                          ===============  =========================  ==========
                          stuff            Array                      [
                                                                        "one",
                                                                        "two"
                                                                      ]

```

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please use psxml to format table https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_format.ps1xml?view=powershell-7.2

File: src/Resources/ResourceManager/SdkModels/Deployments/DeploymentVariable.cs [hunk 1]
```diff
@@ -23,5 +29,17 @@ public class DeploymentVariable
 
         [JsonProperty("value")]
         public object Value { get; set; }
+
+        public override string ToString()
```
Comment (msJinLei) [inline src/Resources/ResourceManager/SdkModels/Deployments/DeploymentVariable.cs:33]: Please use a name that better describes its function, such as ToFormattedString()

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/Resources/ResourceManager/SdkModels/Deployments/DeploymentVariable.cs (hunk 1 above)
Comment (BethanyZhou) [reply src/Resources/ResourceManager/SdkModels/Deployments/DeploymentVariable.cs:33]: Hi @msJinLei , thanks for your comments. I think ToString() is good here as its ToString() only show its type before. No breaking change will introduce here. And I hope user can obtain formatted string anytime.
image

Shall we keep ToString() as the function name?

Comment (msJinLei) [reply src/Resources/ResourceManager/SdkModels/Deployments/DeploymentVariable.cs:33]: > Hi @msJinLei , thanks for your comments. I think ToString() is good here as its ToString() only show its type before. No breaking change will introduce here. And I hope user can obtain formatted string anytime. image
> 
> Shall we keep ToString() as the function name?

The function contains the logical to control its display format rather serialize it to string. It is better to have those who use it realize it by its name.

**Review by msJinLei — APPROVED** [review]

**Pull request merged**